chili-epfl/FROG
# New engine

A lot of changes.

*TO TEST*:
- go to `Admin` and click `load fake database`
- look at the graph in `Graph Editor`
- go to `Teacher View` and create a session
- Have several students join the session
- Have the teacher click `Start`, then `Next`
- Test activity 1 with the students
- Have the teacher click  `Next`
- Test activity 2 with the students
- ...

*COMMENTS*
- Changes types for specifying the API between engines and activities/operators
- All activities are collaborative (In my opinion it makes no sense to have a test and pass or not the reactive function as props) but can not use the collaborative function if they want
- Most of the activity/operator packages will not work anymore and should be tested and rewritten
- The jigsaw activity and operator, do not do jigsaw yet, but only assign random groups and roles, without bothering with the size of the groups and having one student of each role per group. Also the configuration for both should be thought about more deeply.
